### ISSUE TITLE
Replace underlying Fetch API implementation

### ISSUE THREAD

**Issue by stefan-guggisberg** [adobe/helix-fetch#20]
**Is your feature request related to a problem? Please describe.**
[fetch-h2](https://github.com/grantila/fetch-h2), the current underlying Fetch API implementation, has a serious scalability bottleneck: Parallel requests to the same HTTP/2 origin are effectively serialised, thus defeating the purpose of HTTP/2 Multiplexing. See #14.

**Describe the solution you'd like**
A. Replacing `fetch-h2` with another Fetch API implementation which transparently supports HTTP/1(.1) and HTTP/2 and which provides better scalability for parallel requests. [node-fetch-h2](https://github.com/MikeRalphson/node-fetch-h2/tree/h2) seems to be a good candidate.

B. Alternatively, implement the relevant parts of the Fetch API and write a simple abstraction of the builtin `http`, `https` and `http2` modules tailored to our needs.

**Describe alternatives you've considered**
I've created a `fetch-h2` [issue](https://github.com/grantila/fetch-h2/issues/85) and looked into creating a PR. The code however is hardly documented and the flow hard to follow. The maintainer doesn't seem to be very responsive either.

**Comment by stefan-guggisberg** [comment]
Update: The underlying `fetch-h2` [issue](https://github.com/grantila/fetch-h2/issues/85) has been fixed in the meantime.  However, I still think it's worthwhile writing our own Fetch API implementation on top of our own `http`, `https` and `http2` abstraction.

**Comment by tripodsan** [comment]
hurah!
